image_client.py

```diff
@@ -141,54 +160,61 @@ def upload_to_image_server(self, full_path, redacted, collection, original_path=
             r = requests.get(self.build_url("getfileref"), params=params)
             url = r.text
             assert r.status_code == 200
+
             logging.info(f"Uploaded: {local_filename},{attach_loc},{url}")
             logging.info("adding to image")
+
             self.monitoring_tools.add_imagepath_to_html(image_path=original_path,
                                                         barcode=remove_non_numerics(os.path.basename(local_filename)),
                                                         success=True)
 
-        logging.debug("Upload to file server complete")
-
+        self.logger.debug("Upload to file server complete")
 
         return url, attach_loc
 
     # works for just basename +ext. "exact" does a sql "like" operation
     def check_image_db_if_filename_imported(self, collection, filename, exact=False):
         params = {
-            'file_string': filename,
+            'file_string': quote(filename),
             'coll': collection,
             'exact': exact,
             'search_type': 'filename',
-            'token': self.generate_token(filename)
+            'token': self.generate_token(quote(filename))
         }
 
         return self.decode_response(params)
 
-    def write_image_metadata(self, exif_dict, collection, filename):
+
```

Comment:
for consistency with the other image_client calls which directly input the method name into the function self.build_url , maybe not worth it to define the api_target as a separate variable above.